src/components/OpenFileDialog/OpenFileDialog.js

```diff
@@ -0,0 +1,444 @@
+                                                                ) ? (
+                                                                    <span className="info-text">
+                                                                        {getString(
+                                                                            type,
+                                                                            'noFilteredDataText'
+                                                                        )}
+                                                                    </span>
+                                                                ) : (
+                                                                    <>
+                                                                        <div className="info-text">
+                                                                            {getString(
+                                                                                type,
+                                                                                'noDataText'
+                                                                            )}
+                                                                        </div>
+                                                                        <div className="info-button">
+                                                                            <Button
+                                                                                onClick={() => {
+                                                                                    onNew()
+                                                                                    onClose()
+                                                                                }}
+                                                                            >
+                                                                                {getString(
+                                                                                    type,
+                                                                                    'newButtonLabel'
+                                                                                )}
+                                                                            </Button>
+                                                                        </div>
+                                                                    </>
+                                                                )}
+                                                            </div>
+                                                        </div>
+                                                    </Box>
+                                                </DataTableCell>
+                                            </DataTableRow>
+                                        )}
+                                    {Boolean(
```

Comment:
won't this literally type "false" in the ui if the condition isn't met?
use a ternary with a null fallback instead?

Reply:
Actually, this works perfectly fine (new to me!), but since we use the ternary in other places it might be best to just follow that path

Reply:
Rewritten removing the casting and testing for `length > 0` instead.